pages/api/system/reload.js

```diff
@@ -234,28 +234,68 @@ export default async function handler(req, res) {
       }
 
       // - events
-      try {
-        if (profile.events) {
+      async function getCoordinates(city, state, country) {
+        let locationDb = {};
+        const provided = [city, state, country].filter((x) => x).join(",");
+        if (locationDb[provided]) {
+          return locationDb[provided];
+        }
+        try {
+          const location = await fetch(
+            `https://nominatim.openstreetmap.org/?addressdetails=1&q=
+            ${encodeURIComponent(provided)}&format=json&limit=1`
+          );
+          const coordinates = await location.json();
```

Comment:
For events that are added by forms, how do they get their location?

Reply:
We are fetching coordinates with city, state and country details. As soon as we hit the api to load data into MongoDB, it takes the events with `startDate>today` or `EndDate>today` and then fetches coordinates with city, state and country details.
This is what is happening, but I have not taken a look at how forms are working.

Reply:
Ok thanks 👍 Forms will write directly to the db so will skip the `reload` script - I will look into this as I refactor the location fetching with users location